### PR TITLE
Site Editor Hub: Simplify

### LINKED ISSUE #60875: Site Editor: Empty header while in distraction-free mode
### Description

While editing the site in full screen and distraction free mode, the header is empty when hovered. 

This is only happening in the site editor. Post editor is behaving as expected. 

I bisected the behavior change to have been introduced in https://github.com/WordPress/gutenberg/commit/3ece636f6d447f99a0dc8b5def72df548eadb031 

### Step-by-step reproduction instructions

1. Activate Twenty Twenty Four
2. No other plugins
3. Checkout and build commit 3ece636f6d
4. Visit the site editor `/wp-admin/site-editor.php?canvas=edit`
5. Drag the sidebar away (full-screen mode)
6. Turn on "distraction free" mode
7. Hover over header 

Then you can re-build with the parent commit a7a62f740a, and it should work as expected. 

### Screenshots, screen recording, code snippet

https://github.com/WordPress/gutenberg/assets/7129409/786ed302-aa07-40e4-a474-96bb3c59066f

### Environment info

- WordPress 6.5.2 stable
- Chrome Version 123.0.6312.124 (Official Build) (arm64)
- Gutenberg trunk 

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by youknowriad**
## What and Why?

Currently the site editor renders a "hub" with a site logo, site title and some actions. That hub is persistent and rendered as "fixed" on top of everything.

I think the main reason it's implemented that way is to be able to animate the site logo with go in and out of view mode. That said, I personally believe this is way too complex to accommodate for a small animation.

So this PR is kind of a provocation, how simple can it be if we don't have to implement the animation of the logo. It's not necessarily meant to be merged but at least I want to raise this discussion.

Also, coincidentally, this fixes #60875

Let me know what you think.

### CODE REVIEW

File: packages/edit-site/src/components/layout/index.js
Comment (youknowriad) [inline packages/edit-site/src/components/layout/index.js:210]: I think we can probably simplify more here in terms of styles and classes.

File: packages/edit-site/src/components/header-edit-mode/index.js
Comment (youknowriad) [inline packages/edit-site/src/components/header-edit-mode/index.js]: The other good thing here is that this makes the header identical to the post editor and unifying this button is now possible.

**Comment by annezazu** [comment]
Quick note that I tried to use WP playground to test this but I can't see the icon when I'm in the canvas to get a better sense of it:

https://github.com/WordPress/gutenberg/assets/26996883/1f5547e0-c5d8-4082-9203-aed53a4b70cf

**Comment by youknowriad** [comment]
@annezazu looks like a bug, let me check.

**Comment by annezazu** [comment]
Thank you! Here's a little before/after video to show the differences:

https://github.com/WordPress/gutenberg/assets/26996883/eae2555a-0b8d-4d64-9b21-fe7d0e67962c

Here's the prior PR afaik that animated this: https://github.com/WordPress/gutenberg/pull/48886 I do think some level of magic and polish is lost with this animation gone. Thinking ahead to more personalization efforts with the admin redesign, I imagine this same site "hub" will continue to play a big role to orient folks and some level of animation will go a long way to convey a smoothness and cohesiveness of the experience.

**Comment by youknowriad** [comment]
I agree that the animation is nice but what I'm saying is that the way it's implemented today (keeping the same DOM element) is way too complex for what it adds and unless we find a better way to do it, it's not worth it. An example of this is the distraction free bug here https://github.com/WordPress/gutenberg/issues/60875

I think "view transitions" would be perfect here. They already work on Chromium browsers (Chrome, Edge) but not yet in Safari and Firefox. It can be done as a progressive enhancement though.

**Comment by youknowriad** [comment]
I pushed a quick prototype for view transitions in Chrome. You can see now in Chrome that the "site icon" is animated. The animation is slightly different than what we have in trunk (I believe we can make it similar to trunk if we really want to) but I'm really excited about this. It's huge improvement over the gymnastic we need to do in our code base to implement animations today.

Also for site hub specifically, it can be acceptable to leverage this as a progressive enhancement: works on Chromium browsers (more than 71% of all users) while giving up on the hub animation for the rest for now and keeping the simplicity of our code base and fixing the distraction free mode bug.

**Comment by annezazu** [comment]
Tested again and it doesn't feel very smooth or polished compared to before:

https://github.com/WordPress/gutenberg/assets/26996883/f1194f86-d790-414b-a7cf-e62a9847faaa

This is especially the case when returning to the overall navigation where the site icon zooms in almost too much. When entering the canvas, it feels delayed. I'm not sure if there's a way to make it more seamless while still getting the benefits of this PR as I definitely want to see this bug resolved https://github.com/WordPress/gutenberg/issues/60875. 

As is though, this PR isn't quite there.

**Comment by youknowriad** [comment]
I updated this PR to delay only one side of the animation. I think this is way better. Thoughts?

**Review by draganescu — COMMENTED** [review]
Distraction Free Mode is still broken with this in Firefox:


https://github.com/WordPress/gutenberg/assets/107534/ac288f2f-ab6c-49e4-84d1-caf58eea7b17

The animation works very well in other browsers.

**Comment by youknowriad** [comment]
@draganescu  what's broken exactly in Firefox? For me it's working, so I wonder if we're talking about a different unrelated bug.

**Comment by draganescu** [comment]
@youknowriad as the video shows in Firefox when the edit mode is entered and DFM is on the  interface header contains no buttons and stays there empty.

On this PR in other browsers the current bug in trunk where the header does not appear at all in DFM is fixed. So the fix is incomplete somehow in FF and this PR introduces the problem in FF.

**Comment by youknowriad** [comment]
@draganescu I can reproduce that bug in any browser, it depends on where you click to enter "edit mode". I suspect that it's another bug unrelated with the current fix but I'll take a look anyway.

**Comment by youknowriad** [comment]
Ok, so the bug is due to multiple header animations conflicting with each other, I solved it by simplifying and unifying the header animation in a single place.

**Review by jeryj — COMMENTED** [review]
We'll need a way to manage focus between the site hub and the header. On `trunk`, the button for each is the same, so we don't need to do anything to move focus. On this, we're removing the clicked button and it doesn't look like we're placing focus anywhere.

**Comment by youknowriad** [comment]
@jeryj I'm guessing that you're thinking about the move from "edit" to "view" mode right? because for the opposite direction, it's probably the same as trunk.

**Comment by jeryj** [comment]
Yes, edit to view mode. I forgot that going from view to edit, you enter by selecting the canvas, which places focus on the canvas correctly.

**Comment by youknowriad** [comment]
Added focus handling when leaving edit mode.

**Review by draganescu — COMMENTED** [review]
I am unsure how to deal with this PR. One one hand the PR fixes the bug in `trunk` which makes the DFM impossible to exit in the site editor. On the other hand the UI of DFM is severely changed in terms of animation:

- there is a noticeably bigger delay on the hover effect
- the animation is just a simple fade, a big contrast to the rest of the editor UI

I think it's a better overall experience offered by this PR, given that in chrome the hub animation is nearly identical to `trunk` and the other browsers show some minute differences. But scoped only to DFM it's worse.

As a note, I remember Matias was pretty precise and persistent on how the hover animation of DFM. I wasn't even able to land a change that would use CSS transitions to clean up some of the ~~nesting hell~~ complexity introduced by motion components. The animation had to:

- be timed correctly
- be orchestrated correctly
- have the background fade in
- have the hub slide from the left, the left tools from the top and the right tools from the right

... given the bug in `trunk` maybe this PR is OK and should land and then strive to fix the UI for DFM during the WP beta cycle? cc @jasmussen

**Comment by jasmussen** [comment]
> ... given the bug in trunk maybe this PR is OK and should land and then strive to fix the UI for DFM during the WP beta cycle?

Yes, that sounds good to me.

**Comment by youknowriad** [comment]
> should land and then strive to fix the UI for DFM during the WP beta 

@draganescu Can you explain more what are the remaining DFM issues in this PR?

**Comment by draganescu** [comment]
> Can you explain more what are the remaining DFM issues in this PR?

I thought I did, extensively, but here is some video 😁 

WP 6.5.4-alpha, no Gutenberg:

https://github.com/WordPress/gutenberg/assets/107534/164ed7e4-6765-4326-9b71-e2fe1c03c74a

WP 6.5.4-alpha, with Gutenberg on this PR:


https://github.com/WordPress/gutenberg/assets/107534/7ccc237f-c072-4588-9a5b-d2e6f2de850b



Differences:

- the hub does not slide from the left
- the tools don't slide from the top
- it all fades
- there is an increased delay hovering the header until the animation kicks in
- there is a loss of the bounced fade out of the header on mouse out

**Comment by youknowriad** [comment]
Thanks for sharing the details. I agree they are different but my opinion is that the animation in this PR is better. Obviously it's very arguable and subjective. What is objective is that in terms of implementation and code base, it's way simpler (so better).

For the delays, I actually did it on purpose to delay the hover effect but I didn't delay it on leave (which I can do if needed). I'm happy to tweak the delays.

Would love some design opinion on the DFM animation.

**Comment by jasmussen** [comment]
I don't think the difference in animation is a meaningful differential to block this PR. Though I do agree it can be improved, I'd argue that's more about animating the whole top-bar in from the top, rather than fading it in.

**Comment by youknowriad** [comment]
I don't want to argue too much about the animation, so to unblock this PR (and another PR :P ) I extracted some of the animation changes into this smaller PR https://github.com/WordPress/gutenberg/pull/62167 

Let's try to land this related PR first. It touches the header animation but hopefully try to stay as close as possible to the trunk animation. Would love reviews there.

**Comment by youknowriad** [comment]
Now that #62167 landed, I've rebased this PR. So now the DFM bug if fixed in the site editor and the animation is correct as well.

**Comment by draganescu** [comment]
Small bug in this PR compared to trunk is that toolbars remain visible when going to view mode:

This PR:


https://github.com/WordPress/gutenberg/assets/107534/8292b235-a4eb-439d-a9a2-758ccf906fa0




Trunk:


https://github.com/WordPress/gutenberg/assets/107534/e46d9e64-19af-400b-bd90-b644422ab6f8

**Comment by richtabor** [comment]
This tests well (especially in Chromium browsers). I did note the same find that @draganescu pointed out.

File: packages/edit-site/src/components/layout/style.scss
Comment (richtabor) [inline packages/edit-site/src/components/layout/style.scss]: It's tiny, but `225ms` gives the frame enough time to move over before scaling back the site logo.

File: packages/edit-site/src/components/site-hub/index.js
Comment (richtabor) [inline packages/edit-site/src/components/site-hub/index.js:66]: Perhaps it's ok, but there is a bit of a difference in site icon size from trunk: 

CleanShot 2024-05-31 at 08 12 50

Simply increasing the scale doesnt work great though, as then the WP SVG also gets increased slightly.

**Comment by richtabor** [comment]
It's close. There's still a bit of a flash of the site logo/WP SVG (odd few frames) as the editor header is moving out, particularly when you're exiting edit mode. 

https://github.com/WordPress/gutenberg/assets/1813435/683f43f5-c45d-488e-be16-f5bfab7aa48e

https://github.com/WordPress/gutenberg/assets/1813435/78c6a2bf-10f6-4f5d-aebd-101870f70903

**Comment by youknowriad** [comment]
- Rebased the PR. I'm not able to reproduce the toolbars being visible anymore (but let me know if I'm not testing properly).
- I adjusted the delay like above.
- I'm failing to understand the flash of the W svg: Any help there would be appreciated.

**Comment by richtabor** [comment]
> I'm failing to understand the flash of the W svg: Any help there would be appreciated.

I shortened the clip to see it here: 

https://github.com/WordPress/gutenberg/assets/1813435/2fca6556-787d-412b-80ff-99359093a427

**Comment by youknowriad** [comment]
@richtabor I did see the flashing as well, I just don't understand why it's happening :P

**Comment by draganescu** [comment]
> Rebased the PR. I'm not able to reproduce the toolbars being visible anymore (but let me know if I'm not testing properly).

I can still repro both Chrome and FF. Rebuilt everything and still this:



https://github.com/WordPress/gutenberg/assets/107534/17dfa246-32e9-48a4-b4ab-e273599537d0

As you can see not all toolbars remain just top element's ... main group, header template part ...

**Comment by youknowriad** [comment]
@draganescu I fixed the lingering block toolbar.

**Comment by youknowriad** [comment]
@richtabor I think I understand the "W" animation issue a little bit better. The issue comes from the fact that when using the "w" logo or a random site icon, the proportions of the "w" logo are not the same inside and outside the frame. Basically when we're in the editor, we add a padding around the icon but this padding don't exist in view mode which means the two icons actually don't match between edit and view mode, so when we move between one and the other, there's a crossfade animation that is slightly visible because of that padding.

I'm not sure how to solve it without removing that padding (which increases the sizes of the icon in the editor) or adding it to both view and edit mode (which make the icon smaller in view mode).

For sites with an actual image, the proportions are the same.

File: packages/edit-site/src/store/private-actions.js
Comment (draganescu) [inline packages/edit-site/src/store/private-actions.js]: When top toolbar is ON and we exit edit mode to view mode there is a short moment when the header is removing the top toolbar and _then_ the animation moves forward. Can we do it in an unblocking way? Should it be already unblocking? 🤷🏻

**Review by draganescu — APPROVED** [review]
The ghost toolbars are fixed. Left a small problem which I consider a nit.

From my perspective this PR is a good step forward in terms of improving the code base. Given it preserves the way the UI behaved in terms of both interaction and feedback I think this can land given the tests pass. @richtabor do you think the W issue is blocking?

File: packages/edit-site/src/store/private-actions.js
Comment (youknowriad) [reply packages/edit-site/src/store/private-actions.js]: What do you mean "unblocking way"?

Comment (youknowriad) [reply packages/edit-site/src/store/private-actions.js]: I'm not sure that's entirely what you meant but I noticed that we're triggering a big number of actions when switching canvas mode, so I batched everything in 6c58262cf3

I think the animation is smoother with that.

Comment (draganescu) [reply packages/edit-site/src/store/private-actions.js]: Yes this is exactly what I meant :) The animation doesn't look like it's waiting on something to be over before starting, it just goes now. 👏🏻

**Comment by youknowriad** [comment]
Would love another opinion here @richtabor @jasmussen or @mtias before merging this one.

**Comment by stokesman** [comment]
I'm here to invite anyone still subscribed to have a look at #62386 which follows up on this by restoring the previous transition. To me, the transition after this is too degraded to overlook. I tried various ways of fixing it and that’s what seems best to me.

**Comment by ellatrix** [comment]
This was cherry-picked to the wp/6.6 branch.

**Pull request merged**
